### PR TITLE
Map properties features + New UI

### LINKED ISSUE #12: Map properties
**Is your feature request related to a problem? Please describe.**
It could be fun to add some properties for a map (eg: Sliding on ice).
Maybe it should be an another game mode (Classic (without)/ Challenger) ?

**Describe the solution you'd like**
Find a method for the implementation & more effects

- [x] Sliding on Ice

**Additional context**
Should be done for *Milestone 1.1.0*.

### PR DESCRIPTION
**Pull request by AlexisAG**
#12

### CODE REVIEW

File: RedDeadOxygen/Assets/Scripts/Instances/Player.cs [hunk 1]
```diff
@@ -32,10 +32,22 @@ public class Player : MonoBehaviour
     private Timer _speedTimer;
     private List<Ressource> _Ressources = new List<Ressource>();
 
+    private MovementCategory _movementCategory = MovementCategory.e_ForceBased;
+    private float _wallElasticity = 0f;
+
+    Vector2 movement, movementOrder;
+    float _maxForceSpeed = 4.0f, _currentMaxForceSpeed = 4.0f;
```
Comment (AlexisAG) [inline RedDeadOxygen/Assets/Scripts/Instances/Player.cs]: Declare variable on multi-lines

```suggestion
    float _maxForceSpeed = 4.0f, 
    float _currentMaxForceSpeed = 4.0f;
```

File: RedDeadOxygen/Assets/Scripts/Instances/Player.cs [hunk 2]
```diff
@@ -32,10 +32,22 @@ public class Player : MonoBehaviour
     private Timer _speedTimer;
     private List<Ressource> _Ressources = new List<Ressource>();
 
+    private MovementCategory _movementCategory = MovementCategory.e_ForceBased;
+    private float _wallElasticity = 0f;
+
+    Vector2 movement, movementOrder;
```
Comment (AlexisAG) [inline RedDeadOxygen/Assets/Scripts/Instances/Player.cs]: Multi-Line + name convention (be careful, variable reference should be changed too)
```suggestion
    Vector2 _movement; 
    Vector2 _movementOrder;
```

File: RedDeadOxygen/Assets/Scripts/Instances/Player.cs [hunk 3]
```diff
@@ -32,10 +32,22 @@ public class Player : MonoBehaviour
...
+
     public bool PowerUpCooldown = false;
     public Base PlayerBase { get; private set; }
     public GameObject MinePrefab => _minePrefab;
 
+    public enum MovementCategory
+    {
+        e_Basic,
+        e_ForceBased
+    }
```
Comment (AlexisAG) [inline RedDeadOxygen/Assets/Scripts/Instances/Player.cs:51]: Should it be a boolean (?)

File: RedDeadOxygen/Assets/Scripts/Instances/Player.cs [hunk 4]
```diff
@@ -167,7 +196,18 @@ private void Update()
...
+            case MovementCategory.e_Basic:
+                MoveWithController(m_joystickNumber);
+                break;
+            case MovementCategory.e_ForceBased:
+                MoveWithControllerBasedOnForce(m_joystickNumber);
+                break;
+            default:
+                MoveWithController(m_joystickNumber);
+                break;
+        }
```
Comment (AlexisAG) [inline RedDeadOxygen/Assets/Scripts/Instances/Player.cs]:
```suggestion
        switch (_movementCategory)
        {
            case MovementCategory.e_ForceBased:
                MoveWithControllerBasedOnForce(m_joystickNumber);
                break;
            case MovementCategory.e_Basic:
            default:
                MoveWithController(m_joystickNumber);
                break;
        }        
```

**Review by zelectrik — APPROVED** [review]

**Pull request merged**